frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/flavor.tsx

```diff
@@ -87,13 +93,21 @@ export const Flavor: React.FC<FlavorProps> = React.memo(
           <FormPFSelect
             onSelect={(e, v) => nullOnEmptyChange(onChange, VMSettingsField.FLAVOR)(v.toString())}
           >
-            {flavors.map((f) => {
-              return (
-                <SelectOption key={f} value={f}>
-                  {_.capitalize(f)}
-                </SelectOption>
-              );
-            })}
+            {flavors
+              .map(Flavor.fromString)
+              .sort((a, b) => a.getOrder() - b.getOrder())
+              .map((f) => {
+                return (
+                  <SelectOption
+                    key={f.getValue()}
+                    value={f.getValue()}
+                    description={f.getDescription()}
+                  >
+                    {f.toString()}
+                    {f !== Flavor.CUSTOM && ` - ${flavorDescriptions?.[f.getValue()]}`}
```

Comment:
let's resolve the value first and then ask if we found it, so we don't end up with dangling ` - `